spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -367,6 +373,62 @@ public void setPreserveTimestamp(boolean preserveTimestamp) {
 		this.preserveTimestamp = preserveTimestamp;
 	}
 
+	/**
+	 * String setter for Spring XML convenience.
+	 * @param chmod permissions as an octal string e.g "600";
+	 * @see #setChmod(int)
+	 * @since 5.0
+	 */
+	public void setChmodOctal(String chmod) {
+		Assert.notNull(chmod, "'chmod' cannot be null");
+		setChmod(Integer.parseInt(chmod, 8));
+	}
+
+	/**
+	 * Set the file permissions after uploading, e.g. 0600 for
+	 * owner read/write. Only applies to file systems that support posix
+	 * file permissions.
+	 * @param chmod the permissions.
+	 * @since 5.0
+	 */
+	public void setChmod(int chmod) {
+		Assert.isTrue(chmod <= 0777, "'chmod' must be between 0 and 0777 (octal)");
+		BitSet bits = BitSet.valueOf(new byte[] { (byte) chmod, (byte) (chmod >> 8) });
```

Comment:
Great! OK. 

Seems for me I see what you do here.
The `byte[]` has only 8 bits, and with the `chmod >> 8` you get access to the 9'th bit of the `chmod` variable. Where `Integer.toBinaryString(Integer.valueOf("0777", 8)) == "111111111"`